apps/passport-client/components/modals/AccountImportModal.tsx

```diff
@@ -0,0 +1,187 @@
+import { Spacer } from "@pcd/passport-ui";
+import { PCD } from "@pcd/pcd-types";
+import { useCallback, useEffect, useState } from "react";
+import styled from "styled-components";
+import { useFilePicker } from "use-file-picker";
+import { useDispatch, usePCDCollection } from "../../src/appHooks";
+import { Button } from "../core";
+
+type ImportState =
+  | {
+      valid: true;
+      imported: false;
+      pcds: PCD[];
+      folders: FolderMap;
+      conflicts: number;
+    }
+  | { valid: true; imported: true; added: number }
+  | { valid: false };
+
+type FolderMap = { [pcdId: string]: string };
+
+export function AccountImportModal() {
+  const { openFilePicker, filesContent } = useFilePicker({
+    accept: ".json",
+    multiple: false
+  });
+
+  const pcdCollection = usePCDCollection();
+  const dispatch = useDispatch();
+
+  const [importState, setImportState] = useState<ImportState | undefined>();
+
+  const importPCDs = useCallback(() => {
+    // Should never happen, but makes TypeScript happy that we checked for it
+    if (!importState.valid || importState.imported === true) return;
+
+    const addedPCDs: PCD[] = [];
+    for (const pcd of importState.pcds) {
+      if (!pcdCollection.hasPCDWithId(pcd.id)) {
+        try {
+          pcdCollection.add(pcd);
+          if (importState.folders[pcd.id]) {
+            pcdCollection.setFolder(pcd.id, importState.folders[pcd.id]);
+          }
+
+          addedPCDs.push(pcd);
+        } catch (e) {
+          console.log(e);
+        }
+      }
+
+      setImportState({ valid: true, imported: true, added: addedPCDs.length });
+    }
+  }, [importState, pcdCollection]);
+
+  useEffect(() => {
+    (async () => {
+      if (filesContent.length > 0) {
+        try {
+          const file = filesContent[0];
+          const data = JSON.parse(file.content);
+
```

Comment:
You used serializeStorage() to export the file, so I think you should use deserializeStorage() to parse it.  That should deal properly with versioning and be more likely to be kept up-to-date with changes in future.  Is there a reason that's not suitable here?